bundles/io/org.eclipse.smarthome.io.monitor/src/main/java/org/eclipse/smarthome/io/monitor/internal/EventLogger.java

```diff
@@ -7,24 +7,50 @@
  */
 package org.eclipse.smarthome.io.monitor.internal;
 
-import org.eclipse.smarthome.core.events.AbstractEventSubscriber;
-import org.eclipse.smarthome.core.types.Command;
-import org.eclipse.smarthome.core.types.State;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.core.events.Event;
+import org.eclipse.smarthome.core.events.EventFilter;
+import org.eclipse.smarthome.core.events.EventSubscriber;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class EventLogger extends AbstractEventSubscriber {
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Maps;
+
+public class EventLogger implements EventSubscriber {
+
+    private final Map<String, Logger> eventLoggers = Maps.newHashMap();
+
+    private final Set<String> subscribedEventTypes = ImmutableSet.of(EventSubscriber.ALL_EVENT_TYPES);
 
-    private final Logger logger = LoggerFactory.getLogger("runtime.busevents");
+    @Override
+    public Set<String> getSubscribedEventTypes() {
+        return subscribedEventTypes;
+    }
 
     @Override
-    public void receiveCommand(String itemName, Command command) {
-        logger.info("{} received command {}", itemName, command);
+    public EventFilter getEventFilter() {
+        return null;
     }
 
     @Override
-    public void receiveUpdate(String itemName, State newStatus) {
-        logger.info("{} state updated to {}", itemName, newStatus);
+    public void receive(Event event) {
+        Logger logger = getLogger(event.getType());
+        logger.trace("Received event of type '{}' under the topic '{}' with payload: '{}'", event.getType(),
+                event.getTopic(), event.getPayload());
+        logger.info(event.toString());
+    }
+
+    private Logger getLogger(String eventType) {
+        String loggerName = "runtime.busevents." + eventType;
```

Comment:
Not directly related to this PR, but may I ask you to change this to "smarthome." + eventType? Makes the log better readable. Remember to also adapt the logback_debug.xml.